signalling/src/handlers/mod.rs

```diff
@@ -0,0 +1,1081 @@
+use webrtcsink_protocol as p;
+use anyhow::{anyhow, Error};
+use futures::prelude::*;
+use futures::ready;
+use pin_project_lite::pin_project;
+use std::collections::{HashMap, HashSet, VecDeque};
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tracing::{info, instrument};
+
+pin_project! {
+    #[must_use = "streams do nothing unless polled"]
+    pub struct Handler {
+        #[pin]
+        stream: Pin<Box<dyn Stream<Item=(String, Option<p::IncomingMessage>)> + Send>>,
+        items: VecDeque<(String, p::OutgoingMessage)>,
+        producers: HashMap<String, HashSet<String>>,
+        consumers: HashMap<String, Option<String>>,
+        listeners: HashSet<String>,
+    }
+}
+
+impl Handler {
+    #[instrument(level = "debug", skip(stream))]
+    /// Create a handler. When extending the protocol, pass in an
+    /// always pending stream, then call handle() and consume items until
+    /// pending
+    pub fn new(
+        stream: Pin<Box<dyn Stream<Item = (String, Option<p::IncomingMessage>)> + Send>>,
```

Comment:
Maybe you want to add a `Default` impl that uses a `Pending` stream?

Reply:
perhaps in the future, but for now I have unpubbed handle() and removed mentions of it in the comment